### PR TITLE
Pull request validation pipeline

### PR DESCRIPTION
**Pull request by p3dr0rv**
* Migrate classic pipeline to YAML  (Pull Request Validation)
* add spotbugs task

### CODE REVIEW

**Review by iamgusain — APPROVED** [review]

File: azure-pipelines/pull-request-validation/pr-msal.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+- name: BuildParameters.javaHomeSelection
+  value: JDKVersion
+- name: robolectricSdkVersion
+  value: 28
+
+trigger:
+  branches:
+    include:
+    - dev
+    - master
```
Comment (AdamBJohnsonx) [inline azure-pipelines/pull-request-validation/pr-msal.yml:22]: can we add release/*?

File: azure-pipelines/pull-request-validation/pr-msal.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  displayName: Phase 1
+  cancelTimeoutInMinutes: 1
+  pool:
+    name: Hosted Windows 2019 with VS2019
+  steps:
+  - checkout: self
+    clean: true
+    submodules: recursive
+    persistCredentials: True
+  - template: azure-pipelines/templates/steps/credscan-policheck.yml@common
```
Comment (AdamBJohnsonx) [inline azure-pipelines/pull-request-validation/pr-msal.yml:44]: this does use v2, right?

File: azure-pipelines/pull-request-validation/pr-msal.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    parameters:
+      policheckCmdLineArgsDir: msal
+  - template: azure-pipelines/templates/steps/automation-cert.yml@common
+  - task: JavaToolInstaller@0
+    displayName: Use Java 8
```
Comment (AdamBJohnsonx) [inline azure-pipelines/pull-request-validation/pr-msal.yml:49]: How flexible is this java version?  I guess not much, right?

Comment (p3dr0rv) [reply azure-pipelines/pull-request-validation/pr-msal.yml:49]: default value is 8 unless we provide the source
https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/java-tool-installer?view=azure-devops

File: azure-pipelines/pull-request-validation/pr-msal.yml (hunk 2 above)
Comment (p3dr0rv) [reply azure-pipelines/pull-request-validation/pr-msal.yml:44]: yes it does
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/5fbf67bd65f7a122a098328438877ce64646ebf5/azure-pipelines/templates/steps/credscan-policheck.yml#L10

File: azure-pipelines/pull-request-validation/pr-msal.yml (hunk 1 above)
Comment (p3dr0rv) [reply azure-pipelines/pull-request-validation/pr-msal.yml:22]: done here 77ff9ab2fc8560f0006b5c7c0419d27935811452

**Review by AdamBJohnsonx — APPROVED** [review]

**Pull request merged**